plutus-core/testlib/PlutusCore/Generators/PIR/Builtin.hs

```diff
@@ -0,0 +1,59 @@
+{-# LANGUAGE DefaultSignatures #-}
+{-# LANGUAGE TypeApplications  #-}
+
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module PlutusCore.Generators.PIR.Builtin where
+
+import Data.ByteString (ByteString)
+import Data.Coerce
+import Data.Text (Text)
+import Data.Text qualified as Text
+import Data.Text.Encoding qualified as Text
+import Test.QuickCheck
+
+import PlutusCore.Data
+
+instance Arbitrary Data where
+    arbitrary = error "implement me"
+    shrink = error "implement me"
+
+class ArbitraryBuiltin a where
+    arbitraryBuiltin :: Gen a
+    default arbitraryBuiltin :: Arbitrary a => Gen a
+    arbitraryBuiltin = arbitrary
+
+    shrinkBuiltin :: a -> [a]
+    default shrinkBuiltin :: Arbitrary a => a -> [a]
+    shrinkBuiltin = shrink
+
+instance ArbitraryBuiltin ()
+instance ArbitraryBuiltin Bool
+instance ArbitraryBuiltin Integer
+instance ArbitraryBuiltin Data
+
+instance ArbitraryBuiltin Text where
+    arbitraryBuiltin = Text.pack . getPrintableString <$> arbitrary
+    shrinkBuiltin = map (Text.pack . getPrintableString) . shrink . PrintableString . Text.unpack
+
+instance ArbitraryBuiltin ByteString where
+    arbitraryBuiltin = Text.encodeUtf8 <$> arbitraryBuiltin
+    shrinkBuiltin = map Text.encodeUtf8 . shrinkBuiltin . Text.decodeUtf8
+
+newtype AsArbitraryBuiltin a = AsArbitraryBuiltin
+    { unAsArbitraryBuiltin :: a
+    }
+
+instance ArbitraryBuiltin a => Arbitrary (AsArbitraryBuiltin a) where
```

Comment:
What's the point of this? Doing some DerivingVia stuff? Documentation would be good.

Reply:
Yeah, sorry, I [added](https://github.com/input-output-hk/plutus/pull/4559/commits/004587e3fdce614ee04eaf909d8c68f570ddd74c) a bit of docs right after this commit and forgot to mention that.